test/extensions/filters/http/gcp_authn/gcp_authn_filter_integration_test.cc

```diff
@@ -222,6 +235,26 @@ TEST_P(GcpAuthnFilterIntegrationTest, Basicflow) {
   EXPECT_GE(test_server_->counter("cluster.cluster_0.upstream_cx_total")->value(), num);
 }
 
+TEST_P(GcpAuthnFilterIntegrationTest, OverrideDefaultHeader) {
+  initializeConfig(/*add_audience=*/true, true);
+  HttpIntegrationTest::initialize();
+  int num = 2;
+  // Send multiple requests.
+  for (int i = 0; i < num; ++i) {
+    initiateClientConnection();
+    // Send the request to cluster `gcp_authn`.
+    waitForGcpAuthnServerResponse();
+    // Send the request to cluster `cluster_0` and validate the response.
+    sendRequestToDestinationAndValidateResponse(/*with_audience=*/true, true);
```

Comment:
nit : please add annotation for code readability, so that people don't need to check the function definition to figure out which field is set to true.
`/*configure_token_header=*/true`